openmetadata-ui/src/main/resources/ui/src/components/common/TestConnection/TestConnection.tsx

```diff
@@ -0,0 +1,340 @@
+
+// 2 minutes
+const FETCHING_EXPIRY_TIME = 2 * 60 * 1000;
+const FETCH_INTERVAL = 2000;
+const WORKFLOW_COMPLETE_STATUS = [
+  WorkflowStatus.Failed,
+  WorkflowStatus.Successful,
+];
+
+const TestConnection: FC<TestConnectionProps> = ({
+  isTestingDisabled,
+  formData,
+  serviceCategory,
+  connectionType,
+  serviceName,
+  displayButtonOnly = false,
+}) => {
+  const { t } = useTranslation();
+
+  // local state
+  const [isTestingConnection, setIsTestingConnection] =
+    useState<boolean>(false);
+  const [dialogOpen, setDialogOpen] = useState<boolean>(false);
+
+  const [message, setMessage] = useState<string>(
+    t('message.test-your-connection-before-creating-service')
+  );
+
+  const [testConnectionStep, setTestConnectionStep] = useState<
+    TestConnectionStep[]
+  >([]);
+
+  const [testConnectionStepResult, setTestConnectionStepResult] = useState<
+    TestConnectionStepResult[]
+  >([]);
+
+  const [currentWorkflow, setCurrentWorkflow] = useState<Workflow>();
+  const [testStatus, setTestStatus] = useState<TestStatus>();
+
+  /**
+   * Current workflow reference
+   */
+  const currentWorkflowRef = useRef(currentWorkflow);
+
+  // derived variables
+  const updatedFormData = useMemo(() => {
+    return formatFormDataForSubmit(formData);
+  }, [formData]);
+
+  const serviceType = useMemo(() => {
+    return getTestConnectionType(serviceCategory);
+  }, [serviceCategory]);
+
+  const allowTestConn = useMemo(() => {
+    return shouldTestConnection(connectionType);
+  }, [connectionType]);
+
+  const isTestConnectionDisabled =
+    isTestingConnection || isTestingDisabled || !allowTestConn;
+
+  // data fetch handlers
+
+  const fetchConnectionDefinition = async () => {
+    try {
+      const response = await getTestConnectionDefinitionByName(
+        toLower(connectionType)
+      );
+
+      setTestConnectionStep(response.steps);
+    } catch (error) {
+      // we will not throw error for this API
+    }
+  };
+
+  const getWorkflowData = async (workflowId: string) => {
+    try {
+      const response = await getWorkflowById(workflowId);
+      const testConnectionStepResult = response.response?.steps ?? [];
+
+      setTestConnectionStepResult(testConnectionStepResult);
+
+      setCurrentWorkflow(response);
+
+      return response;
+    } catch (error) {
+      throw error as AxiosError;
+    }
+  };
+
+  // handlers
+  const handleTestConnection = async () => {
+    setIsTestingConnection(true);
+    setMessage(t('message.testing-your-connection-may-take-two-minutes'));
+
+    // reset states for workflow ans steps result
+    setCurrentWorkflow(undefined);
+    setTestConnectionStepResult([]);
+    setTestStatus(undefined);
+
+    // current interval id
+    let intervalId: number | null = null;
+
+    try {
+      const createWorkflowData: CreateWorkflow = {
+        name: `test-connection-${connectionType}-${cryptoRandomString({
+          length: 8,
+          type: 'alphanumeric',
```

Comment:
Let's create a util function to get name based on `connectionType`

Reply:
Sure.